WDL/runtime/task.py

```diff
@@ -802,6 +818,61 @@ def map_paths(v: Value.Base, dn: str) -> Value.Base:
     )
 
 
+def link_outputs_relative(
+    cache: CallCache,
+    outputs: Env.Bindings[Value.Base],
+    run_dir: str,
+    hardlinks: bool = False,
+) -> Env.Bindings[Value.Base]:
+    """
+    link_outputs with [file_io] use_relative_output_paths = true. We organize the links to reflect
+    the generated files' paths relative to their task working directory.
+    """
+    link_destinations = dict()
+
+    def map_path_relative(v: Union[Value.File, Value.Directory]) -> str:
+        target = (
+            v.value
+            if os.path.exists(v.value)
+            else cache.get_download(v.value, isinstance(v, Value.Directory))
+        )
+        if target:
+            real_target = os.path.realpath(target)
+            rel_link = None
+            if path_really_within(target, os.path.join(run_dir, "work")):
+                # target was generated by current task; use its path relative to the task work dir
+                if not os.path.basename(run_dir).startswith("download-"):  # except download tasks
+                    rel_link = os.path.relpath(real_target, os.path.join(run_dir, "work"))
+            elif path_really_within(target, run_dir):
+                # target is an out/ link generated by a call in the current workflow (task or
+                # subworkflow); use the link's path relative to that out/ dir
+                rel_link = os.path.relpath(target, run_dir)
+                p = rel_link.find("/out/")
+                rel_link = rel_link[p + 5 :] if p > 0 else None
+            # if neither of the above cases applies, then fall back to just the target basename
```

Comment:
A case is missing for call cached entries. These also have an outdir. I wonder if it is possible to "know" that the code is taking a path from the cache.

Reply:
@rhpvorderman Good catch, I'll look into this